src/sentry/workflow_engine/processors/delayed_workflow.py

```diff
@@ -71,61 +72,82 @@ def fetch_group_to_event_data(
     return buffer.backend.get_hash(model=model, field=field)
 
 
-def get_dcg_group_workflow_data(
-    worklow_event_dcg_data: dict[str, str]
-) -> tuple[dict[int, set[int]], dict[int, int], list[Any]]:
+def get_dcg_group_workflow_detector_data(
+    workflow_event_dcg_data: dict[str, str]
+) -> tuple[dict[int, set[int]], dict[int, int], dict[int, int], list[Any]]:
     """
-    Parse the data in the buffer hash, which is in the form of {workflow_id}:{event_id}:{dcg_id, ..., dcg_id}:{dcg_type}
+    Parse the data in the buffer hash, which is in the form of {workflow/detector_id}:{event_id}:{dcg_id, ..., dcg_id}:{dcg_type}
     """
 
     dcg_to_groups: dict[int, set[int]] = defaultdict(set)
     dcg_to_workflow: dict[int, int] = {}
+    dcg_to_detector: dict[int, int] = {}
     all_event_data = []
 
-    for workflow_event_dcg, instance_data in worklow_event_dcg_data.items():
+    for workflow_event_dcg, instance_data in workflow_event_dcg_data.items():
         data = workflow_event_dcg.split(":")
-        workflow_id = int(data[0])
+        dcg_type = data[3]  # TODO: use dcg_type to split WHEN and IF DCGs
+
+        dcg_has_detector = dcg_type == DataConditionHandlerType.DETECTOR_TRIGGER
+        workflow_or_detector_id = int(data[0])
         event_id = int(data[1])
         dcg_ids = [int(dcg_id) for dcg_id in data[2].split(",")]
-        # TODO: use dcg_type = data[3] to split WHEN and IF DCGs
         event_data = json.loads(instance_data)
 
         for dcg_id in dcg_ids:
             dcg_to_groups[dcg_id].add(event_id)
-            dcg_to_workflow[dcg_id] = workflow_id
+            if dcg_has_detector:
```

Comment:
```py
target_dict = dcg_to_detector if dcg_type == DataConditionHandlerType.DETECTOR_TRIGGER else dcg_to_workflow
target_dict[dcg_id] = int(data[0])
```
we could simplify some of this logic by getting the target_dict since we always assign it to the same value. (and save those sweet sweet bits of memory on the `workflow_or_detector_id` variable)